### PR TITLE
update vs code extension name and description for seo

### PR DESCRIPTION
**Pull request by chillatom**
## Test plan

Not sure how to test that this displays correctly in marketplace.  Open to suggestions.

### CODE REVIEW

**Review by abeatrix — APPROVED** [review]

File: vscode/package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "cody-ai",
   "private": true,
-  "displayName": "Cody AI",
+  "displayName": "Cody: AI Coding Assistant with Autocomplete & GPT-4 Turbo Chat for JS, Python, Java, Go, & more",
```
Comment (toolmantim) [inline vscode/package.json:4]: I wonder if this also updates the product name in VS Code notifications etc?

Comment (toolmantim) [reply vscode/package.json:4]: I believe this will change this label too:

image

Given that, I'm not sure we should make this particular line change.

Comment (toolmantim) [reply vscode/package.json:4]: I'll check where else within the product UI this label appears, just so we know how to balance this.

Comment (chillatom) [reply vscode/package.json:4]: Good catch! Any way to work around this? Like an internal name variable that is different from the marketplace display name?

File: vscode/package.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   "version": "1.10.0",
   "publisher": "sourcegraph",
   "license": "Apache-2.0",
   "icon": "resources/cody.png",
-  "description": "Code AI with codebase context",
+  "description": "Cody is an AI coding assistant that understands your codebase. Cody brings autocomplete, chat, and commands to your IDE, helping you generate code, write unit tests, create documentation, and explain complex or legacy code. Choose from several LLMs including GPT-3.5 Turbo, GPT-4 Turbo, and Claude 3. Cody supports all major languages including JavaScript, TypeScript, Python, Java, Go, and C/C++.",
```
Comment (abeatrix) [inline vscode/package.json:9]:
```suggestion
  "description": "AI coding assistant that understands your codebase. Cody brings autocomplete, chat, and commands to your IDE, helping you generate code, write unit tests, create documentation, and explain complex or legacy code using AI. Choose from several LLMs including GPT-3.5 Turbo, GPT-4 Turbo, and Claude 3. Cody supports all major languages including JavaScript, TypeScript, Python, Java, Go, and C/C++.",
```

**Pull request closed without merging**